oshmem/mca/spml/ucx/spml_ucx.c

```diff
@@ -571,7 +571,7 @@ void mca_spml_ucx_rmkey_free(sshmem_mkey_t *mkey, int pe)
     if (!mkey->spml_context) {
         return;
     }
-    segno = memheap_find_segnum(mkey->va_base);
+    segno = memheap_find_remote_segnum(mkey->va_base, pe);
```

Comment:
Is it only invoked for remote segments?

Reply:
What if it’s not found?

Reply:
For rmkey_free it should invoke memheap_find_remote_segnum for remote segments. For local segments, it should invoke usual memheap_find_segnum in deregister. 
va_base in rmkey_free is fetched from mkey using the same map which we are using to find the seg number. So, in rmkey_free flow it should always give the valid segno. Invalid segno is returned if it fails to find the va_address in the same map. Do you suggest incorporating a error check here ?